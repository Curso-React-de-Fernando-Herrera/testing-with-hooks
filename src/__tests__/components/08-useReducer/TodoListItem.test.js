import React from 'react'
import { shallow } from 'enzyme'
import { TodoListItem } from 'components/08-useReducer/TodoListItem'
import { demoTodo } from '__tests__/fixture/demoTodo'

// todo, index, handleDelete, handleToggle
describe('TodoListItem component', () => {

    const handleDelete = jest.fn()
    const handleToggle = jest.fn()

    const wrapper = shallow(
      <TodoListItem
        todo= { demoTodo[0] }
        index= {1}
        handleDelete={handleDelete}
        handleToggle= {handleToggle}
      />
    )

  test('should be render', () => {
    expect( wrapper ).toMatchSnapshot()
  })

  test('should be call handleToggle', () => {
    wrapper.find('p').simulate('click')

    expect( handleToggle ).toHaveBeenCalledWith(1)
  })

  test('should be call handleDelete', () => {
    wrapper.find('button').simulate('click')

    expect( handleDelete ).toHaveBeenCalledWith(1)
  })

  test('should be call correct text', () => {
    const text = wrapper
      .find('p')
      .text()
      .trim()

    expect( text ).toBe(`${2}. ${demoTodo[0].desc}`)
  })

  test('should be have class complete if it TODO.done = true', () => {
    const newTodo = demoTodo[0]
    newTodo.done = true

    const wrapper = shallow(
      <TodoListItem
        todo= { newTodo }
      />
    )

    expect( wrapper.find('p').hasClass('complete') ).toBe( true )
    expect( wrapper.find('p').hasClass('complete') ).not.toBe( false )
  })
})