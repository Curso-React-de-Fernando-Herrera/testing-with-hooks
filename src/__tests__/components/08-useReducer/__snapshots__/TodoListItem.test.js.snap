// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem component should be render 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    . 
    Hola mundo!
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
